src/AggregateFunctions/SingleValueData.cpp

```diff
@@ -466,22 +468,40 @@ std::optional<size_t> SingleValueDataFixed<T>::getSmallestIndexNotNullIf(
         {
             opt = findExtremeMinNotNull(vec.getData().data(), null_map, row_begin, row_end);
             if (!opt.has_value())
-                return opt;
+                return std::nullopt;
+            T smallest = *opt;
             for (size_t i = row_begin; i < row_end; i++)
             {
-                if (!null_map[i] && vec_data[i] == *opt)
-                    return {i};
+                if constexpr (is_floating_point<T>)
+                {
+                    if (!null_map[i] && std::memcmp(&vec_data[i], &smallest, sizeof(T)) == 0)
```

Comment:
How does `std::memcmp(&vec_data[i], &smallest, sizeof(T)) == 0` work with BFloat16? I'm not sure that it is POD because there were some fixes with that regard recently: https://github.com/ClickHouse/ClickHouse/pull/77290

Reply:
It works ok.

Reply:
I'll add a static_assert to confirm